crates/oxc_ast/src/ast/js.rs

```diff
@@ -1530,6 +1530,7 @@ pub struct Function<'a> {
     pub expression: bool,
     pub generator: bool,
     pub r#async: bool,
+    pub this_param: Option<TSThisParameter<'a>>,
```

Comment:
let's add a documentation for this, a reference to the typescript manual + an example in code.